src/middlewares/authentication.js

```diff
@@ -0,0 +1,60 @@
+import { Request, Response, NextFunction } from 'express';
+import { verify } from 'jsonwebtoken';
+import customMessages from '../utils/customMessages';
+import statusCodes from '../utils/statusCodes';
+import UserService from '../services/authentication.service';
+import responseHandlers from '../utils/responseHandlers';
+
+const { errorResponse } = responseHandlers;
+const { unAuthorized } = statusCodes;
+const {
+  tokenAbsent,
+  tokenVerifyFailed,
+  accountNotVerified,
+} = customMessages;
+
+/**
+   * @description Verifies authenticity of current user
+   */
+export default class Authentication {
+  /**
+   * @param {Request} req Node/Express Request object
+   * @param {Response} res Node/Express Response object
+   * @param {NextFunction} next Node/Express Next callback function
+   * @returns {NextFunction | Object} Node/Express Next callback function or an error response
+   * @description Checks existence of a token in the Authorization request header,
+   * decodes it, and finaly adds the decoded data
+   * to Node/Express's Request Object as sessionUser property
+   */
+  static async isCurrentUserLoggedIn(req, res, next) {
+    let token = req.get('authorization');
+    if (!token) {
+        return errorResponse(res, unAuthorized, tokenAbsent);
+    }
+    token = token.split(' ').pop();
```

Comment:
Can you make sure you are using the OAuth 2.0 standard format of authorization header?

Reply:
could you elaborate @the22mastermind